src/blog/2024/01/capture-data-edge-with-node-red-flowfuse.md

```diff
@@ -0,0 +1,62 @@
+
+<!--more-->
+
+FlowFuse enables data to be collected through Node-RED. Data can be processed locally on the edge or sent on to other services. FlowFuse doesn’t rely on continuous connections to the cloud, making it a good choice for locations with unreliable internet connectivity. Use cases like real-time monitoring of critical systems, proactive maintenance and improved operational efficiency are now possible to implement.
+
+## Installing the FlowFuse agent
+
+To manage the capturing of data on the edge we’re going to first install the FlowFuse agent. It’s installed on your device to manage the communication between the edge device and the FlowFuse server, manage the installation of Node-RED, its execution environment, and facilitate communication between devices and the cloud.
+
+The device agent can run anywhere you can run a Docker container or Node.JS runtime (version 16.0+) can be installed.
+
+
+### Registering a device on FlowFuse
+
+For the edge device to know what it’s supposed to do it needs to listen to the FlowFuse commands. To agents configuration is provided by a `device.yml` file from FlowFuse. Go to the team you’d like to add an edge device to, select “Devices” on the left hand menu followed by the “Add Device” button.
+
+![Setting up a FlowFuse agent](./images/flowfuse-agent-setup.png "Setting up a FlowFuse agent")
+
+FlowFuse will prompt you to add a name (required), and a type (not required). When you’ve clicked `Add` you’ll get a new dialog to download the required file.
+
+![Configuration file for the FlowFuse agent](./images/device-yml-flowfuse.png "The contents of a device.yml file")
+
+### Install the FlowFuse agent through Docker
+
+If your edge device is capable of running containers, it’s the fastest way to run the FlowFuse agent. Assuming you’ve already got Docker installed, we there’s just two steps to follow: first move the device YAML file downloaded from FlowFuse to the edge device and save it in `/opt/flowfuse/device.yml`. Start the agent by running:
+
+Note that for production cases, make sure the container is restarted on reboot. Docker can do this for you, [please follow their guide](https://docs.docker.com/config/containers/start-containers-automatically/).
+
+
+### Install the FlowFuse agent with npm
+
+To install the agent through NPM, you’ll need a Node.JS version of 18.0 or later. Open a command prompt and run:
+
+This will install the FlowFuse Device Agent as a global npm module, making the flowfuse-device-agent command available in any directory on your system.
+
+Once the installation is complete, you need to configure the Device Agent to connect to your FlowFuse instance. In this guide you’ve previously downloaded the `device.yml` file that’s needed now. On linux or mac, move the file to `/opt/flowfuse-device/device.yml`, for Windows based systems to `c:\opt\flowfuse-device\device.yml`.
+
+Afterwards start the agent with:
+
```

Comment:
Are we missing code here to start the agent?

Reply:
```suggestion
```
flowfuse-device-agent
```
```